core/services/kraken/extension/extension.py

```diff
@@ -0,0 +1,301 @@
+import asyncio
+import json
+from typing import AsyncGenerator, Dict, List, Literal, Optional, cast
+
+from commonwealth.settings.manager import Manager
+from loguru import logger
+
+from config import SERVICE_NAME
+from extension.exceptions import (
+    ExtensionNotFound,
+    ExtensionNotRunning,
+    ExtensionPullFailed,
+    IncompatibleExtension,
+)
+from extension.models import ExtensionSource
+from harbor import ContainerManager, DockerCtx
+from harbor.exceptions import ContainerNotFound
+from manifest import ManifestManager
+from manifest.models import ExtensionVersion
+from settings import ExtensionSettings, SettingsV2
+
+
+class Extension:
+    """
+    Extension class to manage extensions.
+    """
+
+    # If an extension is being installed the key will be the extension identifier if is being removed the key is the
+    # container name.
+    locked_entries: Dict[str, Literal[True]] = {}
```

Comment:
Usually I prefer in this kind of lock operation to use dict just to avoid duplicated keys and keep the lock, we could just change to a list with filter also in the unlock, but its just a preference, if you think is better I can change to list.